### ISSUE TITLE
Missing Support for input[type="radio"] and select

### ISSUE THREAD

**Issue by srsgores** [maestrooo/ember-cli-html5-validation#1]
I noticed validation is missing for these inputs.  Are there any plans of implementing?

**Comment by tmaynx** [comment]
Hi guys,

seems like support for select boxes is still not completely working.

My markup looks like:
```
<div class="form-control">
  <select {{action 'change' on='change'}} required={{required}}>
    {{#each content key="@index" as |item|}}
      <option value="{{item.id}}" selected={{eq item.id selected}}>
        {{item.description}}
      </option>
    {{/each}}
   </select>
</div>
```

This markup is inside a component, which gives several params (eg. content, required) to the select box.

In case that "required" is true, i expect that the plugin will validate the select and put an error class on the form-control if no option is selected, but it does not. I am using Ember 1.13.4 with Ember-CLI 1.13.1.

Any solutions for that?

**Comment by bakura10** [comment]
Hi @tmaynx ,

I'm not doing any EmberJS work for now so unfortunately I won't be able to provide an immediate fix, but feel free to submit a PR for that.

Normally, the requirement to work is to make sure that you have an option whose value is empty (value=""). Is that the case?

**Comment by tmaynx** [comment]
Hi @bakura10,

that is unfortunate, but nevermind, I'll see what I can do =)

Problem is that I currently have no idea how I can fix the issue. Seems that the validation listener is not even registered on my select box. If I extend my component (which one contains the select box) with validation mixin, it can't find an appropriate input respectively select element.

To answer your question: The rendered output has an option whose value is empty, requirements are fulfiled here.

**Comment by bakura10** [comment]
Hi,

Ha, ok. I see what happen. You are actually using the <select> tag directly, and not the Select view: http://emberjs.com/api/classes/Ember.Select.html

If you use the select view it will work, however I know there is less flexibility in that.

If I rmemeber correctly the Select view is kinda deprecated in Ember but currently this is the only way to make the validation triggered on select.

For now, the philosophy of this library has been to have a completely drop-in solution: just use the standard input component, select view... and it will do the validaiton. However this is way less flexible (hard to format errors the way you want, for instance), so I think that in the future, we'll need to use our own component, like {{validatable-input}}...

**Comment by tmaynx** [comment]
I see. Unfortunately i can't use the Ember Select view, because we don't want to have any deprecations in our project. That means I have to find a way to trigger the validation on my select element without using the view.

Thank you for your support and the explanations so far. I'm send you a PR when I have figured it out =)

**Comment by bakura10** [comment]
After checking more it seems that the {{view "select"}} is not deprecated in 1.x branch

Envoyé de mon iPhone

**Comment by srsgores** [comment]
How about the ``ember-x-select`` add-on?

**Comment by bakura10** [comment]
Yeah, sounds like a nice idea. Anyway I'll wait until we have EmberJS 2, there are a lot of new featuers that may be of interest to such a library like this one!
